driver-core/src/main/com/mongodb/internal/connection/PowerOfTwoBufferPool.java

```diff
@@ -51,27 +110,37 @@ public PowerOfTwoBufferPool() {
      * @param highestPowerOfTwo the highest power of two buffer size that will be pooled
      */
     public PowerOfTwoBufferPool(final int highestPowerOfTwo) {
+        this(highestPowerOfTwo, DEFAULT_MAX_IDLE_TIME_MINUTES, TimeUnit.MINUTES);
+    }
+
+    /**
+     * Construct an instance.
+     *
+     * @param highestPowerOfTwo the highest power of two buffer size that will be pooled
+     * @param maxIdleTime max idle time when pruning is enabled
+     * @param timeUnit time unit of maxIdleTime
+     */
+    public PowerOfTwoBufferPool(final int highestPowerOfTwo, final long maxIdleTime, final TimeUnit timeUnit) {
```

Comment:
It appears that this constructor can be package-access. If that's the case, let's remove `public`.

Reply:
This led me down a small rabbit hole.  Since besides the test, all clients must use `DEFAULT`, then those clients should not be able to enable pruning or close `DEFAULT`.  So I also made the pruning methods package-access, and removed `AutoCloseable` from the interface list.